lib/fs.js

```diff
@@ -2008,12 +2008,12 @@ function ReadStream(path, options) {
   this.closed = false;
 
   if (this.start !== undefined) {
-    if (typeof this.start !== 'number') {
+    if (typeof this.start !== 'number' || Number.isNaN(this.start)) {
       throw new ERR_INVALID_ARG_TYPE('start', 'number', this.start);
     }
     if (this.end === undefined) {
       this.end = Infinity;
-    } else if (typeof this.end !== 'number') {
+    } else if (typeof this.end !== 'number' || Number.isNaN(this.end)) {
```

Comment:
Can you add this for the other `typeof this.end !== 'number'` check below as well? The one that is supposed to be merged with this one? :)

Reply:
@addaleax Sure! On it.

Reply:
@addaleax done.